src/client/react/components/CustomFunctionsDashboard/List.tsx

```diff
@@ -55,27 +62,64 @@ export interface Item {
 export default ({ items }) => (
   <ListContainer>
     {items.map((item: Item) => (
-      <ListItem
-        key={item.key}
-        className="ms-font-s"
-        backgroundColor={item.background}
-        color={item.color}
-        title={item.title}
-        smallCaps={item.smallCaps}
-      >
-        {item.icon && (
-          <Icon
-            className="ms-font-m"
-            iconName={item.icon.name}
-            style={{
-              fontSize: '16px',
-              color: item.icon.color,
-              marginRight: '5px',
-            }}
-          />
-        )}
-        <Text style={{ marginLeft: (item.indent || 0) * 5 + 'px' }}>{item.name}</Text>
-      </ListItem>
+      <div id={item.name}>
+        <ListItem
+          key={item.key}
+          className="ms-font-s"
+          id={item.name}
+          backgroundColor={item.background}
+          color={item.color}
+          title={item.title}
+          smallCaps={item.smallCaps}
+        >
+          <Text>
+            {item.dropdown && (
+              <Icon
+                className="ms-font-m"
+                iconName={item.dropdown.name}
+                style={{
+                  width: '18px',
+                  fontSize: '16px',
+                  color: item.dropdown.color,
+                  marginRight: '5px',
+                  marginLeft: (item.indent || 0) * 5 + 'px',
+                }}
+                onClick={() =>
```

Comment:
Couple issues with this:
1) by directly putting a lambda here, it will recreate this lambda each time the render is called. If it is rendered 1000 times, this function will get created 1000 times. You can fix this by binding the function in the constructor, or making the lambda a member variable of the class (with arrow functions) and using that instead.
2) you'll want to avoid manually manipulating the dom with React, especially when it comes to visibility. Instead, choose to conditionally render such as seen [here](https://github.com/OfficeDev/MakerIDE/blob/master/src/components/Backstage/GalleryList/index.tsx#L32)